### PR TITLE
chore: cleanup and simplify examples

### PR DESCRIPTION
**Pull request by amrbashir**
- Updated `.gitignore` file
- Simplified and cleanedup examples
- Removed examples that serve almost zero value:
  - `navigation`
  - `parent-window`
  - `web`
  - `workspace`
  - `plugins/tauri-plugin-sample`: the plugin example in `api` example is enough

### CODE REVIEW

File: examples/api/src-tauri/Cargo.toml [hunk 1]
```diff
@@ -37,14 +37,3 @@ features = [
 [dev-dependencies.tauri]
 path = "../../../core/tauri"
 features = ["test"]
-
-[features]
-prod = ["tauri/custom-protocol"]
```
Comment (lucasfernog) [inline examples/api/src-tauri/Cargo.toml:42]: hey i liked this feature.. a super convenient way to run the app in production mode

File: examples/run-iteration/main.rs [hunk 2]
```diff
@@ -1,26 +0,0 @@
...
-
-fn main() {
-  let mut app = tauri::Builder::default()
-    .build(tauri::generate_context!(
-      "../../examples/run-iteration/tauri.conf.json"
-    ))
-    .expect("error while building tauri application");
-
-  loop {
-    app.run_iteration(|_app, _event| {
```
Comment (lucasfernog) [inline examples/run-iteration/main.rs]: i think we should keep this example.. there's no other usage of this api

File: examples/api/src-tauri/Cargo.toml (hunk 1 above)
Comment (amrbashir) [reply examples/api/src-tauri/Cargo.toml:42]: I didn't find it being used somewhere so just removed it, let's bring it back

File: examples/run-iteration/main.rs (hunk 2 above)
Comment (amrbashir) [reply examples/run-iteration/main.rs]: yeah I was hesitant too, let's bring that back as well

**Comment by amrbashir** [comment]
so linking the api example manifest using `tauri-winres` results in `STATUS_ENTRYPOINT_NOT_FOUND` but if directly embedding it using `cargo:rustc-link-arg=/MANIFEST:EMBED` has no issues. I hate windows

**Comment by lucasfernog** [comment]
> so linking the api example manifest using `tauri-winres` results in `STATUS_ENTRYPOINT_NOT_FOUND` but if directly embedding it using `cargo:rustc-link-arg=/MANIFEST:EMBED` has no issues. I hate windows

I guess the `__TAURI_WORKSPACE__` workaround doesn't help us here right? since it's a binary that must have that one

**Comment by amrbashir** [comment]
The `__TAURI_WORKSPACE__` workaround also uses `cargo:rustc-link-arg=/MANIFEST:EMBED` and that's what works

**Comment by amrbashir** [comment]
I will spend sometime to test and see what's the different and why this happens.

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**